File: frontend/src/types/project.ts
export type DeploymentStatus = "CREATED" | "BUILDING" | "RUNNING" | "FAILED";

export interface Project {
  id: number;
  repo_name: string;
  status: DeploymentStatus;
  created_at: string;
  repo_id?: number;
  repo_full_name?: string;
  default_branch?: string;
  private?: boolean;
  clone_url?: string;
  updated_at?: string;
}

export interface ProjectDetailResponse {
  id: number;
  repo_name: string;
  repo_full_name: string;
  clone_url: string;
  default_branch: string;
  private: boolean;
  status: DeploymentStatus;
  created_at: string;
  updated_at: string;
}

export interface DeployProjectResponse {
  status: DeploymentStatus;
}

export interface ProjectImportRequest {
  repo_id: number;
}

export interface GitHubRepository {
  id: number;
  name: string;
  private: boolean;
  default_branch: string;
}

File: backend/app/schemas/project.py
from datetime import datetime
from typing import Optional
from pydantic import ConfigDict
from pydantic import BaseModel

class ProjectCreate(BaseModel):
    repo_id: int
    repo_name: str
    repo_full_name: str
    default_branch: str
    private: bool
    clone_url: str


class ProjectImportRequest(BaseModel):
    repo_id: int


class GitHubRepositoryResponse(BaseModel):
    id: int
    name: str
    private: bool
    default_branch: str


class ProjectResponse(BaseModel):
    id: int 
    repo_name: str
    status: str
    created_at: datetime
    deployment_url: Optional[str] = None
    subdomain: Optional[str] = None

    model_config = ConfigDict(from_attributes=True )


class ProjectDetailResponse(BaseModel):
    id:int
    repo_name:str
    repo_full_name: str
    clone_url:str
    default_branch:str
    private: bool
    status: str
    deployment_url: Optional[str] = None
    subdomain: Optional[str] = None
    created_at: datetime
    updated_at: datetime

    model_config = ConfigDict(from_attributes=True)
